Task3/Project/DataControl/Services/CsvFileService.cs

```diff
@@ -1,53 +1,301 @@
+        /// Basic constructors without parameters.
+        /// </summary>
+        public CsvFileService()
+        {
+            rand = new System.Random();
+        }
+
+        // METHODS
+        private void CheckFileConfiguration()
+        {
+            if (fileConfiguration == null)
+            {
+                throw new System.ArgumentNullException("The configuration isn`t setted up.");
+            }
+        }
+
+        private string[] GetRandomSplittedLine(string path)
+        {
+            string[] lines = File.ReadAllLines(path);
+            int index = rand.Next(0, lines.Length);
+
+            return lines[index].Split(';');
+        }
+
+        private TaxiDriver.Client CreateClient()
+        {
+            string[] clientParameters = GetRandomSplittedLine(fileConfiguration.ClientFile);
+            return new TaxiDriver.Client(clientParameters[1], clientParameters[clientParameters.Length - 1]);
+        }
+
+        private TaxiDriver.Route CreateRoute()
+        {
+            string[] routeParameters = GetRandomSplittedLine(fileConfiguration.RouteFile);
+
+            int startStreetId = int.Parse(routeParameters[1]);
+            int endStreetId = int.Parse(routeParameters[2]);
+
+            System.Tuple<string, string> startAndEndStreets = GetStartAndEndStreets(startStreetId, endStreetId);
+
+            return new TaxiDriver.Route(startAndEndStreets.Item1,
+                startAndEndStreets.Item2,
+                System.TimeSpan.FromSeconds(int.Parse(routeParameters[3])),
+                double.Parse(routeParameters[routeParameters.Length - 1]));
+        }
+
+        private System.Tuple<string, string> GetStartAndEndStreets(int startStreetId, int endStreetId)
+        {
+            string line, startStreet = null, endStreet = null;
+            using (StreamReader streamReader = new StreamReader(fileConfiguration.StreetFile))
+            {
+                int maxIndex = startStreetId > endStreetId ? startStreetId : endStreetId;
+                int i = 0;
+                while (i < maxIndex)
+                {
+                    line = streamReader.ReadLine();
+                    if (i == startStreetId - 1)
+                    {
+                        startStreet = line.Split(';')[1];
+                    }
+                    else if (i == endStreetId - 1)
+                    {
+                        endStreet = line.Split(';')[1];
+                    }
+                    ++i;
+                }
+            }
+            return new System.Tuple<string, string>(startStreet, endStreet);
+        }
+
+        private void SetDriverTotalAndBestScore()
+        {
+            string[] scoreLines = File.ReadAllLines(fileConfiguration.ScoreFile);
+
+            for (int i = 0; i < scoreLines.Length; ++i) 
+            {
+                if (scoreLines[i].Contains(driver.Name))
+                {
+                    driver.LastScore = double.Parse(scoreLines[i].Split(';')[1]);

```

Comment:
weird

Reply:
method names thet it return Best and Total Score, but the Last Score is setted

Reply:
not return best and total score, but SET them

Reply:
looks at implementation of "LastScore" property from "Driver" class, it is a good feature.

Reply:
i am not arguing, if you think that its work than ok, just test it out